contrib/insights/src/column_analyzer.cxx

```diff
@@ -0,0 +1,120 @@
+        
+        auto dt = static_cast<mlio::dense_tensor*>(t.get());
+        auto cells = dt->data().as<std::string>();
+
+        auto should_capture = capture_columns->count(feature_idx) > 0;
+
+        // Used to update the mean.
+        // We do this here to aggregate several entries together and avoid
+        // potential numerical problems updating the mean a single entry at a time. 
+        double numeric_column_sum = 0.0;
+        double numeric_column_count = 0.0;
+
+        for (std::string as_string : cells) {
+            feature_statistics.rows_seen += 1;
+            feature_statistics.example_value = as_string;
+            
+            if (should_capture) {
+                if (feature_statistics.string_captured_unique_values.size() < max_capture_count) {
+                    feature_statistics.string_captured_unique_values.insert(as_string);
+                } else if (feature_statistics.string_captured_unique_values.count(as_string) == 0) {
+                    // If the value isn't present but we're not adding it because we're at a limit
+                    // then we should flag that we have overflowed.
+                    feature_statistics.string_captured_unique_values_overflowed = true;
+                }
+            }
+
+            // String analyzers.
+            feature_statistics.string_cardinality_estimator.add(as_string);
+            
+            feature_statistics.string_min_length = 
+                std::min(feature_statistics.string_min_length, as_string.size());
+            
+            feature_statistics.string_max_length = 
+                std::max(feature_statistics.string_max_length, as_string.size());
+
+            if (as_string.size() == 0) {
+                feature_statistics.string_empty_count += 1;
+            } else {
+                feature_statistics.string_min_length_not_empty = 
+                    std::min(feature_statistics.string_min_length_not_empty, as_string.length());
+            }
+
+            if (mlio::only_whitespace(as_string)) {
+                feature_statistics.string_only_whitespace_count += 1;
+            }
+
+            auto lowercase_string = as_string;
+            std::transform(as_string.begin(), as_string.end(), lowercase_string.begin(), ::tolower);
+            if (mlio::matches(lowercase_string, null_like_values)) {
+                feature_statistics.string_null_like_count += 1;
+            }
+
+            // Numeric analyzers
+            double as_float;
+            if (mlio::try_parse_float(mlio::v1::float_parse_params{as_string}, as_float) != mlio::parse_result::ok) {
+                feature_statistics.numeric_nan_count += 1;
+            } else {
+                feature_statistics.numeric_count += 1;
+                numeric_column_sum += as_float;
+                numeric_column_count += 1;
+                if (isnan(feature_statistics.numeric_min) || as_float < feature_statistics.numeric_min) {
```

Comment:
Use `std::isnan` instead of `isnan`. This line in this form fails to compile on Ubuntu 18.04. Most likely the list of included headers in this compilation unit also needs to be cleaned up.